examples/name-service/programs/name-service/src/lib.rs

```diff
@@ -0,0 +1,249 @@
+    },
+    InstructionDataInvokeCpi, NewAddressParamsPacked, OutputCompressedAccountWithPackedContext,
+};
+use light_utils::hash_to_bn254_field_size_be;
+
+declare_id!("7yucc7fL3JGbyMwg4neUaenNSdySS39hbAk89Ao3t1Hz");
+
+#[program]
+pub mod name_service {
+
+    use light_sdk::utils::create_cpi_inputs_for_new_address;
+
+    use super::*;
+
+    pub fn create_record<'info>(
+        ctx: Context<'_, '_, '_, 'info, NameService<'info>>,
+        proof: CompressedProof,
+        new_address_params: NewAddressParamsPacked,
+        name: String,
+        rdata: RData,
+        cpi_context: Option<CompressedCpiContext>,
+    ) -> Result<()> {
+        let record = NameRecord {
+            owner: ctx.accounts.signer.key(),
+            name,
+            rdata,
+        };
+        let compressed_account = create_compressed_account(&ctx, &record, &new_address_params)?;
+
+        let signer_seed = b"cpi_signer".as_slice();
+        let bump = Pubkey::find_program_address(&[signer_seed], &ctx.accounts.self_program.key()).1;
+        let signer_seeds = [signer_seed, &[bump]];
+
+        let inputs = create_cpi_inputs_for_new_address(
+            proof,
+            new_address_params,
+            compressed_account,
+            &signer_seeds,
+            cpi_context,
+        );
+
+        verify(ctx, &inputs, &[&signer_seeds])?;
+
+        Ok(())
+    }
+
+    pub fn update_record<'info>(
+        ctx: Context<'_, '_, '_, 'info, NameService<'info>>,
+        compressed_account: PackedCompressedAccountWithMerkleContext,
+        proof: CompressedProof,
+        address: [u8; 32],
+        name: String,
+        rdata: RData,
+        cpi_context: Option<CompressedCpiContext>,
+    ) -> Result<()> {
+        signer_check(&ctx, &compressed_account)?;
+
+        let record = NameRecord {
+            owner: ctx.accounts.signer.key(),
+            name,
+            rdata,
+        };
+        let new_compressed_account = compressed_account_with_address(&record, address)?;
+
+        let signer_seed = b"cpi_signer".as_slice();
+        let bump = Pubkey::find_program_address(&[signer_seed], &ctx.accounts.self_program.key()).1;
+        let signer_seeds = [signer_seed, &[bump]];
+
+        let inputs = InstructionDataInvokeCpi {
+            proof: Some(proof),
+            new_address_params: vec![],
+            input_compressed_accounts_with_merkle_context: vec![compressed_account],
+            output_compressed_accounts: vec![new_compressed_account],
+            relay_fee: None,
+            compress_or_decompress_lamports: None,
+            is_compress: false,
+            signer_seeds: signer_seeds.iter().map(|seed| seed.to_vec()).collect(),
+            cpi_context,
+        };
+
+        verify(ctx, &inputs, &[&signer_seeds])?;
+
+        Ok(())
+    }
+
+    pub fn delete_record<'info>(
+        ctx: Context<'_, '_, '_, 'info, NameService<'info>>,
+        compressed_account: PackedCompressedAccountWithMerkleContext,
```

Comment:
same here the input account is not hashed onchain.

Reply:
Done.